app/objects/secondclass/c_relationship.py

```diff
@@ -56,7 +56,7 @@ def shorthand(self):
                 stub += f" : {self.target.name}({self.target.value})"
         return stub
 
-    def __init__(self, source, edge=None, target=None, score=1, origin=None):
+    def __init__(self, source=None, edge=None, target=None, score=1, origin=None):
```

Comment:
Much like with Facts, Relationships are defined as having, at the bare minimum, 1 source fact - without it, there's no way to actually interact with/search for the Relationship. Is there some error caused by this not having a default value that we've overlooked with the API upgrades?